tests/simulator_test.cpp

```diff
@@ -132,25 +142,29 @@ TEST_F(EnergyCollectorTest, NotEvenNumOfEnergyCollectorTest) {
 
   Ray alongX(kVecZero, kVecX);
   ASSERT_TRUE(performHitCollector(energyCollectors, alongX, &hitData));
-  ASSERT_EQ(collectorPositionRadius - refCollectorRadius, hitData.time);
+  ASSERT_FLOAT_EQ(collectorPositionRadius - refCollectorRadius, hitData.time);
 
   Ray alongY(kVecZero, kVecY);
   ASSERT_TRUE(performHitCollector(energyCollectors, alongY, &hitData));
-  ASSERT_EQ(collectorPositionRadius - refCollectorRadius, hitData.time);
+  ASSERT_FLOAT_EQ(collectorPositionRadius - refCollectorRadius, hitData.time);
 
-  Ray at2Angle(kVecZero, Vec3(std::cos(2 * collectorAngle), 0,
-                              std::sin(2 * collectorAngle)));
+  const float collectorAngle = getCollectorAngle(energyCollectors);
+  Ray at2Angle(kVecZero,
```

Comment:
Code reuse. There is still code in those tests that you have duplicated, and could be made into a function.

Reply:
done